stress/shell/src/main/java/alluxio/stress/cli/MultiOperationStressMasterBench.java

```diff
@@ -0,0 +1,502 @@
+        operationIndex = operationIndexSelector.apply(operationIndex);
+        operationCount = mContext.getOperationCounter(operationIndex).incrementAndGet() - 1;
+        long startNs = System.nanoTime();
+        applyOperation(operationIndex, operationCount);
+        long endNs = System.nanoTime();
+
+        long currentMs = CommonUtils.getCurrentMs();
+        // Start recording after the warmup
+        if (currentMs > recordMs) {
+          mResult.incrementNumSuccess(operationIndex, 1);
+
+          // record response times
+          long responseTimeNs = endNs - startNs;
+          mResponseTimeNs[operationIndex].recordValue(responseTimeNs);
+
+          // track max response time
+          long[] maxResponseTimeNs = mResult.getStatistics(operationIndex).mMaxResponseTimeNs;
+          int bucket =
+              Math.min(maxResponseTimeNs.length - 1, (int) ((currentMs - recordMs) / bucketSize));
+          if (responseTimeNs > maxResponseTimeNs[bucket]) {
+            maxResponseTimeNs[bucket] = responseTimeNs;
+          }
+        }
+      }
+    }
+
+    /**
+     * Gets the ratio array and normalizes it into probability distribution
+     * e.g. [4,6,10] will be normalized into [0.2, 0.3, 0.5]
+     * @return the probability distribution array
+     */
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private double[] getProbabilityDistribution(double[] ratios) {
+      double ratioSum = 0;
+      double last = 0;
+      double[] probabilities = new double[ratios.length];
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        ratioSum += mParameters.mOperationsRatio[i];
+      }
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        probabilities[i] = last + mParameters.mOperationsRatio[i] / ratioSum;
+        last = probabilities[i];
+      }
+      return probabilities;
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runOperationsRatioBased() throws Exception {
+      double[] probabilities = getProbabilityDistribution(mParameters.mOperationsRatio);
+      //Each run randomly picks an operation based on the probability array
+      runInternal(lastOperationIndex -> {
+        double r = mRandom.nextDouble();
+        int operationIndex = 0;
+        for (int i = 0; i < mParameters.mOperations.length; i++) {
+          if (probabilities[i] >= r) {
+            operationIndex = i;
+            break;
+          }
+        }
+        mContext.getRateLimiter(0).acquire();
+        return operationIndex;
+      });
+    }
+
+    @SuppressFBWarnings("BC_UNCONFIRMED_CAST")
+    private void runThreadsRatioBased() throws Exception {
+      double[] probabilities = getProbabilityDistribution(mParameters.mThreadsRatio);
+      int operationIndex = 0;
+      //Each thread randomly picks an operation to do based on the probability array
+      for (int i = 0; i < mParameters.mOperations.length; i++) {
+        if (probabilities[i] >= mThreadIndex) {
```

Comment:
because thread index is an integer and probabilities are < 1

```suggestion
        if (probabilities[i] >= mThreadIndex / totalThreads) {
```

Reply:
done